src/main/java/org/jenkinsci/plugins/GithubAuthenticationToken.java

```diff
@@ -103,13 +103,20 @@ public GithubAuthenticationToken(String accessToken, String githubServer) throws
 
         this.userName = this.me.getLogin();
         authorities.add(SecurityRealm.AUTHENTICATED_AUTHORITY);
-        Map<String, Set<GHTeam>> myTeams = gh.getMyTeams();
-        for (String orgLogin : myTeams.keySet()) {
-            LOGGER.log(Level.FINE, "Fetch teams for user " + userName + " in organization " + orgLogin);
-            authorities.add(new GrantedAuthorityImpl(orgLogin));
-            for (GHTeam team : myTeams.get(orgLogin)) {
-                authorities.add(new GrantedAuthorityImpl(orgLogin + GithubOAuthGroupDetails.ORG_TEAM_SEPARATOR
-                        + team.getName()));
+        if(Jenkins.getInstance().getSecurityRealm() instanceof GithubSecurityRealm) {
+            GithubSecurityRealm myRealm = (GithubSecurityRealm) Jenkins.getInstance().getSecurityRealm();
+            String[] myScopes = myRealm.getOauthScopes().split(",");
+            Arrays.sort(myScopes);
+            if(Arrays.binarySearch(myScopes, "read:org") >= 0 || Arrays.binarySearch(myScopes, "admin:org") >= 0 || Arrays.binarySearch(myScopes, "user") >= 0) {
```

Comment:
This conditional needs a comment.
If I were to come across this line while working on the code, I wouldn't have any idea what's being checked here.
To figure it out, I'd have to reach for the GitHub API docs and work backwards from "What are the common properties of these scopes?"

Probably something like "Check to see if we can read organizational and team memberships" would do the trick